thriftbp/client_middlewares.go

```diff
@@ -390,6 +400,44 @@ func PrometheusClientMiddleware(remoteServerSlug string) thrift.ClientMiddleware
 	}
 }
 
+func FaultInjectionClientMiddleware(address string) thrift.ClientMiddleware {
+	return func(next thrift.TClient) thrift.TClient {
+		return thrift.WrappedTClient{
+			Wrapped: func(ctx context.Context, method string, args, result thrift.TStruct) (thrift.ResponseMeta, error) {
+				if address == "" {
+					return next.Call(ctx, method, args, result)
+				}
+
+				var getHeaderFn faults.GetHeaderFn = func(key string) string {
```

Comment:
nit:
```suggestion
				getHeaderFn := func(key string) string {
```

Reply:
same for the other 2 lambdas and the httpbp middleware.

Reply:
Done.